### PR TITLE
ci: add markdown linter

### PR DESCRIPTION
**Pull request by nathan-weinberg**
can run locally with make md-lint

### CODE REVIEW

**Comment by nathan-weinberg** [comment]
lot of Markdown Lint errors here - I'll go through them over time but too many to do in one fell swoop myself immediately

**Comment by ckadner** [comment]
> lot of Markdown Lint errors here - I'll go through them over time but too many to do in one fell swoop myself immediately

Looks like a lot of trailing white spaces. Should be possible to use basic tools, editors and even plain git to do those fixes. I'm willing to help, but don't want to duplicate any work you already did?

**Comment by nathan-weinberg** [comment]
Haven't start anything @ckadner so if you want to push commits onto this PR be my guest! Appreciate it!

**Comment by bjhargrave** [comment]
I updated the linter addition commit and also fixed the linting problems in the repo.

**Comment by nathan-weinberg** [comment]
@bjhargrave looks like the bots are getting territorial :wink: 

joking aside - thanks a lot for this, much appreciated!

**Comment by bjhargrave** [comment]
I am not sure the Makefile is useful since one can npm or brew install the markdownlint-cli2 command for local linting. The container image is awkward to use on macOS.

One can also install the [VSCode markdownlint extension](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) for live linting in their editor.

**Review by russellb — APPROVED** [review]

**Comment by russellb** [comment]
> I am not sure the Makefile is useful since one can npm or brew install the markdownlint-cli2 command for local linting. The container image is awkward to use on macOS.
> 
> One can also install the [VSCode markdownlint extension](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) for live linting in their editor.

I use it frequently on mac. It seems to work fine for me. It's obviously not required to use, though.

File: README.md [hunk 1]
```diff
@@ -33,19 +34,21 @@ Learn about the concepts of "skills" and "knowledge" in our [InstructLab Communi
...
 > The skill taxonomy structure is used in several ways:
->    1. To select the right subset of the taxonomy to use for data generation.
->    2. To determine the interpretability by human contributors and maintainers.
->    3. As part of the prompt to the GPT model used to generate synthetic samples.
-
+>
+> 1. To select the right subset of the taxonomy to use for data generation.
+> 2. To determine the interpretability by human contributors and maintainers.
+> 3. As part of the prompt to the LLM used to generate synthetic samples.
+<!-- -->
```
[the remark this answers is not in the document]
Comment (bjhargrave) [reply README.md:51]: https://github.com/DavidAnson/markdownlint/blob/v0.34.0/doc/md028.md

**Comment by russellb** [comment]
also another thing on the `Makefile`, it replicates the markdown lint toolchain used in several other instructlab repos. @nathan-weinberg are talking about setting up some automation for the files common across repos to make it easy to keep updates in sync, but it's not a "this week" kind of thing.

**Pull request merged**